### DISCUSSION TITLE
Assign an IP address to VLAN

### DISCUSSION THREAD

**Discussion by andrewm659** [netbox-community/netbox#9341] [General]
Should I create an interface on the switch and call it VLAN_XYZ and assign it an  IP address?  Or create a VLAN and can I assign an IP address there?

**Comment by candlerb** [comment]
An *interface* can be associated with an *IP address* - e.g. 192.168.1.2 (*)

A *VLAN* can be assigned to a *prefix* - e.g. 192.168.1.0/24.  It means that this VLAN (layer 2 domain) carries this subnet.

-----

(*) In Netbox you have to enter this as 192.68.1.2/24 - i.e. it's a single address but you have to store the interface prefix length as well.

**Comment by CassidyHunt** [comment]
I struggled with the process until I saw this guy on Youtube.  He does a great job of explaining how everything works.

https://www.youtube.com/channel/UC05NSijsZpNM_mbd0-qagLA
